p2p/protocol/ping/ping_test.go

```diff
@@ -0,0 +1,43 @@
+package ping
+
+import (
+	"testing"
+	"time"
+
+	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
+	peer "github.com/ipfs/go-ipfs/p2p/peer"
+	netutil "github.com/ipfs/go-ipfs/p2p/test/util"
+)
+
+func TestPing(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	h1 := netutil.GenHostSwarm(t, ctx)
+	h2 := netutil.GenHostSwarm(t, ctx)
+
+	err := h1.Connect(ctx, peer.PeerInfo{
+		ID:    h2.ID(),
+		Addrs: h2.Addrs(),
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ps1 := NewPingService(h1)
+	_ = NewPingService(h2)
+
+	pctx, cancel := context.WithCancel(ctx)
+	ts, err := ps1.Ping(pctx, h2.ID())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	select {
+	case took := <-ts:
+		t.Log("ping took: ", took)
+	case <-time.After(time.Second * 4):
+		t.Fatal("failed to receive ping")
+	}
+
```

Comment:
i think this test needs to:
- [ ] ping in both directions as the test did before
- [ ] try pinging multiple times in a row